config/notifiers.go

```diff
@@ -492,7 +492,7 @@ func (c *OpsGenieConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 		return err
 	}
 
-	if c.APIURL != nil && len(c.APIKeyFile) > 0 {
+	if c.APIKey != "" && len(c.APIKeyFile) > 0 {
```

Comment:
```suggestion
        if c.APIURL == nil {
          return fmt.Errorf("missing URL in config")
        }
        
	if c.APIKey != "" && len(c.APIKeyFile) > 0 {
```

Reply:
I think we need to validate the URL as well, `DefaultOpsGenieConfig` does not contain a default URL.

Reply:
there's a default OpsGenie URL in the global config.
https://github.com/prometheus/alertmanager/blob/23f961ec6a9c8a15bf99c83815d35e13d66a4d5d/config/config.go#L591

Reply:
🤦